### PR TITLE
Make swish an inlinable function

### PR DESCRIPTION
**Pull request by benbarsdell**
The implementation of swish currently uses a Defun that specifies noinline=True for the purpose of forcing recomputation of the sigmoid(x) tensor, instead of incurring the memory overhead of keeping it alive from the forward pass.

This detail (unique within the codebase) is undesirable because non-inlined functions are difficult to deal with inside graph optimization passes. E.g., the layout optimizer in grappler is currently unable to optimize through a non-inlined function call.

This commit changes the implementation of swish to achieve the same force-recomputation effect using a control dependency instead of setting noinline=True, allowing the function to be inlined and fully optimized by the backend.

Attn. @reedwm 
cc. @nluehr

### CODE REVIEW

**Review by reedwm — DISMISSED** [review]

File: tensorflow/python/ops/nn_impl.py [hunk 1]
```diff
@@ -504,26 +504,27 @@ def _swish_shape(op):
   return [op.inputs[0].shape]
 
 
-@function.Defun(shape_func=_swish_shape, func_name="swish_grad", noinline=True)
+@function.Defun(shape_func=_swish_shape, func_name="swish_grad")
```
Comment (alextp) [inline tensorflow/python/ops/nn_impl.py]: The right answer here is to remove function.Defun altogether and replace its usage of a custom gradient with custom_gradient.custom_gradient.

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/nn_impl.py (hunk 1 above)
Comment (benbarsdell) [reply tensorflow/python/ops/nn_impl.py]: Done. Let me know if it looks OK.

**Review by alextp — DISMISSED** [review]
Thanks!

**Comment by reedwm** [comment]
Please fix test errors

**Review by reedwm — APPROVED** [review]

**Pull request merged**